codegen-server/typescript/src/main/kotlin/software/amazon/smithy/rust/codegen/server/typescript/smithy/generators/TsApplicationGenerator.kt

```diff
@@ -0,0 +1,303 @@
+            """
+            pub struct App {
+                handlers: Handlers,
+            }
+            """,
+        )
+        Attribute("napi").render(writer)
+        writer.rustBlock("impl App") {
+            renderAppCreate(writer)
+            renderAppStart(writer)
+        }
+    }
+
+    private fun renderAppCreate(writer: RustWriter) {
+        Attribute("napi(constructor)").render(writer)
+        writer.rustBlockTemplate(
+            """pub fn create(ts_handlers: TsHandlers) -> #{napi}::Result<Self>""",
+            *codegenScope,
+        ) {
+            operations.map { operation ->
+                val operationName = symbolProvider.toSymbol(operation).name
+                val input = "crate::input::${operationName}Input"
+                val fnName = operationName.toSnakeCase()
+                rustTemplate(
+                    """
+                    let $fnName: #{napi}::threadsafe_function::ThreadsafeFunction<
+                        $input, #{napi}::threadsafe_function::ErrorStrategy::Fatal
+                    > = ts_handlers.$fnName.create_threadsafe_function(0, |ctx| Ok(vec![ctx.value]))?;
+                    """,
+                    *codegenScope,
+                )
+            }
+            rust("let handlers = Handlers {")
+            operations.map { operation ->
+                val operationName = symbolProvider.toSymbol(operation).name.toSnakeCase()
+                rust("    $operationName: $operationName.clone(),")
+            }
+            rust("};")
+            writer.rust("Ok(Self{ handlers })")
+        }
+    }
+
+    private fun renderAppStart(writer: RustWriter) {
+        Attribute("napi").render(writer)
+        writer.rustBlockTemplate(
+            """pub fn start(&self, socket: &TsSocket) -> #{napi}::Result<()>""",
+            *codegenScope,
+        ) {
+            rustTemplate(
+                """
+                // #{SmithyTs}::setup_tracing();
+                let plugins = #{SmithyServer}::plugin::PluginPipeline::new();
+                let builder = crate::service::PokemonService::builder_with_plugins(plugins);
+                """,
+                *codegenScope,
+            )
+            operations.map { operation ->
+                val operationName = symbolProvider.toSymbol(operation).name.toSnakeCase()
+                rust("let builder = builder.$operationName(crate::ts_operation_adaptor::$operationName);")
+            }
+            rustTemplate(
+                """
+                let app = builder.build().expect("failed to build instance of PokemonService")
+                    .layer(&#{SmithyServer}::AddExtensionLayer::new(self.handlers.clone()));
+                let service = #{tower}::util::BoxCloneService::new(app);
+                // #{SmithyTs}::server::start_hyper_worker(socket, service).expect("failed to start the hyper server");
+                start_hyper_worker(socket, service).expect("failed to start the hyper server");
```

Comment:
We're not exporting anything here as opposed to `TsSocket`, is there a reason to not use `#{SmithyTs}::server::start_hyper_worker(socket, service).expect("failed to start the hyper server");`?